src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,582 @@
+     * status
+     *
+     * @return Mapping between plugin name and boolean status
+     */
+    /*
+     * private ArrayList<PluginTrackerModel> readCsv() {
+     * ArrayList<PluginTrackerModel> accumulator = new ArrayList<>();
+     * try {
+     * CSVReader reader = new CSVReader(new FileReader(mainLockFile));
+     * String[] line;
+     * while ((line = reader.readNext()) != null) {
+     * accumulator.add(new PluginTrackerModel(line[0],
+     * Boolean.parseBoolean(line[1]), line[2], line[3]));
+     * }
+     * reader.close();
+     * return accumulator;
+     * } catch (FileNotFoundException e) {
+     * return accumulator;
+     * } catch (CsvException | IOException e) {
+     * throw new RuntimeException(e);
+     * }
+     * }
+     */
+
+    /**
+     * Read a JSON file at the given location file and return a mapping between
+     * plugin names and install
+     * status
+     *
+     * @param csv The mapping between plugin name and boolean status
+     */
+    private ArrayList<PluginTrackerModel> readJson() {
+        try {
+            // ObjectMapper mapper = new ObjectMapper();
+            // Reads the json file and converts it to a list of PluginTrackerModel objects
+            return this.mapper.readValue(new File(mainLockFile), mapper.getTypeFactory()
+                    .constructCollectionType(ArrayList.class, PluginTrackerModel.class));
+        } catch (FileNotFoundException e) {
+            return (new ArrayList<PluginTrackerModel>());
+        } catch (JsonParseException | JsonMappingException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    /**
+     * Save an ArrayList's contents into a JSON file by replacing all contents
+     *
+     * @param list ArrayList of PluginTrackerModel instances to save
+     *
+     */
+    private void saveJson(ArrayList<PluginTrackerModel> list) {
+        try {
+            this.mapper.writeValue(Paths.get(mainLockFile).toFile(), list);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    /*
+     * Save an ArrayList's contents, overwriting a CSV file
+     *
+     * @param list ArrayList of PluginTrackerModel instances to save
+     *
+     */
+    /*
+     * private void saveCsv(ArrayList<PluginTrackerModel> listToSave) {
+     * String csvFile = mainLockFile;
+     * try (CSVWriter writer = new CSVWriter(new FileWriter(csvFile))) {
+     * for (PluginTrackerModel pluginTrackerModel : listToSave) {
+     * String[] data = { pluginTrackerModel.getName(),
+     * pluginTrackerModel.isManual().toString(),
+     * pluginTrackerModel.getVersionId(), pluginTrackerModel.getPluginId() };
+     * writer.writeNext(data);
+     * }
+     * } catch (IOException e) {
+     * throw new RuntimeException(e);
+     * }
+     * }
+     */
```

Comment:
Please don't comment out code. Delete them. You can see the previously deleted code through git history if you ever want to reference something.